pkg/controller/core/workload_controller.go

```diff
@@ -263,35 +283,56 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
 
+		var requeueAfter *time.Duration
+		if wl.Status.RequeueState != nil && wl.Status.RequeueState.RequeueAt != nil {
+			requeueAfter = ptr.To(wl.Status.RequeueState.RequeueAt.Sub(r.clock.Now()))
+		}
+		if requeueAfter != nil && *requeueAfter > 0 {
+			log.V(3).Info("Waiting for backoff to finish", "backoff", *requeueAfter)
+			return reconcile.Result{RequeueAfter: *requeueAfter}, nil
+		}
+
+		if requeueAfter != nil && workload.Status(&wl) == workload.StatusPending {
```

Comment:
```suggestion
		if wl.Status.RequeueState.RequeueAt != nil && workload.Status(&wl) == workload.StatusPending {
```
to better capture the intention that the code is meant to clean the field